pkgs/development/beam-modules/fetch-mix-deps.nix

```diff
@@ -1,8 +1,16 @@
 { stdenvNoCC, lib, elixir, hex, rebar, rebar3, cacert, git }:
 
-{ name, version, sha256, src, mixEnv ? "prod", debug ? false, meta ? { } }:
-
-stdenvNoCC.mkDerivation ({
+{ name
+, version
+, sha256
+, src
+, mixEnv ? "prod"
+, debug ? false
+, buildEnvVars ? { }
```

Comment:
I think the same thing can apply here, we could do like other fetchers (e.g. [GitHub](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchgithub/default.nix#L11) and [GitLab](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchgitlab/default.nix#L22)) and allow overriding everything that way. I'm not sure why in their case, they place it after, though.

Reply:
My understanding is that I am allowing everything to be overridable
https://github.com/NixOS/nixpkgs/pull/112477/files#diff-c0b648507e9e9261c37bbd0386d43e0daa993def72bd032b67b17b24e5e35e8cR20

The only reason i have added buildEnvVars is that they have a little bit of a special role in that they are needed for the fetching of dependencies as well.

Let me know if I misunderstood you.

Reply:
Yes, you you are already allowing everything to be overridable, including environment variables both in attrs, and in `buildEnvVars`. (Note: you could also override something like `configurePhase` in `buildEnvVars`)

The reason I'm suggesting removing `buildEnvVars` is that it is already normal in nixpkgs to override variables directly in the attributes, and it felt weird to me that `buildMix` should be a special case.

Reply:
I see your point. Thank you for mentioning it.

The only reason I did this was because I use it in fetch-mix-deps.

Perhaps this is one more argument to support the fact that fetch-mix-deps is a separate step and should be done by the developper?

Reply:
I think passing fetchMixDeps explicitly might be actually a good idea. For example this allows to easier override the project. It also makes it easier to understand how the mechanics works, while buildRustPackage/buildGoModule are sometimes cumbersome. The disadvantage so is that requires slightly more boilercode but it also allows us to easier replace fetchMixDeps in future with something that is not a fixed input derivation. What do you think?